src/borg/cache.py

```diff
@@ -524,6 +524,7 @@ def legacy_cleanup():
             self.chunks = create_master_idx(self.chunks)
 
     def add_chunk(self, id, chunk, stats, overwrite=False):
+        # TODO: this is doing an async put, callers need to call repo.barrier for async exceptions
```

Comment:
well yeah, but usually that means the operation is failed, i.e. what I wanted to say here is that this is still safe w.r.t. the transactional model used.